Submitted By: Axel Bergerhoff (axelbergerhoff .a.t. compuserve .d.o.t. com)
Date: 2009-07-02
Initial Package Version: 1.9.25
Origin: Axel Bergerhoff
Upstream Status: Not submitted - LFS Specific
Description: Fix for building on systems with kernel 2.6


diff -Naur svgalib-1.9.25.orig/kernel/svgalib_helper/kernel26compat.h svgalib-1.9.25/kernel/svgalib_helper/kernel26compat.h
--- svgalib-1.9.25.orig/kernel/svgalib_helper/kernel26compat.h	2006-05-20 14:17:13.000000000 +0200
+++ svgalib-1.9.25/kernel/svgalib_helper/kernel26compat.h	2009-07-02 17:37:52.000000000 +0200
@@ -10,7 +10,7 @@
 # define PCI_GET_CLASS pci_find_class
 # define PCI_GET_DEVICE pci_find_device
 
-# if defined (PG_chainlock)
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,3)
 #  define my_io_remap_page_range(vma, start, ofs, len, prot) \
 		io_remap_page_range(vma,start,ofs,len,prot)
 # else
@@ -70,6 +70,7 @@
 
 /* These are also not present in 2.6 kernels ... */
 #if (!defined _LINUX_DEVFS_FS_KERNEL_H) || (defined KERNEL_2_6)
+#include <linux/fs.h>
 static inline int devfs_register_chrdev (unsigned int major, const char *name,
                                          struct file_operations *fops)
 {
@@ -77,7 +78,12 @@
 }
 static inline int devfs_unregister_chrdev (unsigned int major,const char *name)
 {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,0)
     return unregister_chrdev (major, name);
+#else
+    unregister_chrdev (major, name);
+    return 0;
+#endif
 }
 #endif
 
@@ -161,3 +171,11 @@
 #ifndef PCI_VENDOR_ID_RENDITION 
 #define PCI_VENDOR_ID_RENDITION               0x1163
 #endif
+
+#ifndef IRQF_SHARED
+# define IRQF_SHARED SA_SHIRQ
+#endif
+
+#ifndef SA_SHIRQ
+# define SA_SHIRQ
+#endif
